### PR TITLE
Refactor 

### PR DESCRIPTION
**Pull request by Jensen96**

### CODE REVIEW

File: Backend/Backend/BLL/EventCtrl.cs [hunk 1]
```diff
@@ -26,21 +27,30 @@ public Event CreateEvent(DALContext ctx, string title, string description, int n
...
 
-        public Event FindById(DALContext ctx, int eventId)
+        public Event FindById(int eventId)
         {
-            return new EventDB(ctx).FindByID(eventId);
+            return eDB.FindByID(eventId);
         }
 
-        public Registration RegisterToEvent(DALContext ctx, Event e, User user)
+        public void RegisterToEvent(Event evnt, User user)
```
Comment (FnuGk) [inline Backend/Backend/BLL/EventCtrl.cs]: Write a comment explaining that the input params event and user is invalid after the call

File: Backend/Backend/DAL/EventDB.cs [hunk 2]
```diff
@@ -7,22 +7,33 @@
...
+                        e = ctx.Events.Add(entity);
+                        ctx.SaveChanges();
+                        ctxTransaction.Commit();
+                        return e;
+                    }
+                    catch (Exception)
+                    {
+
+                        ctxTransaction.Rollback();
+                        return e;
```
Comment (FnuGk) [inline Backend/Backend/DAL/EventDB.cs]: keep throwing the exception up the stack

File: Backend/Backend/DAL/IEventDB.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace DAL
+{
+    interface IEventDB : ICRUD<Event>
+    {
+        Event FindByID(int id);
+        IEnumerable<Event> FindAll();
+        Event Create(Event entity);
+        void Update(Event entity);
+        void Delete(Event entity);
```
Comment (FnuGk) [inline Backend/Backend/DAL/IEventDB.cs]: Does ICRUD<Event> not implicitly make sure we implement all these methods so they do not have to be redeclared?

File: Backend/Backend/PotLogService/Service.cs [hunk 4]
```diff
@@ -12,30 +12,33 @@ namespace PotLogService
     public class Service : IService
     {
         private UnitOfWork UoW = new UnitOfWork();
```
Comment (Arcanedarkness) [inline Backend/Backend/PotLogService/Service.cs]: Der skal ikke være en unit of work

File: Backend/Backend/DAL/RegistrationDB.cs [hunk 5]
```diff
@@ -9,17 +9,28 @@ namespace DAL
...
+                    try
+                    {
+                        reg = ctx.Registrations.Add(entity);
+                        ctx.SaveChanges();
+                        ctxTransaction.Commit();
+                        return reg;
+                    }
+                    catch (Exception)
+                    {
+                        ctxTransaction.Rollback();
```
Comment (FnuGk) [inline Backend/Backend/DAL/RegistrationDB.cs]: throw the exception up the stack

File: Backend/Backend/BLL/UserCtrl.cs [hunk 6]
```diff
@@ -11,8 +11,8 @@ namespace BLL
     public class UserCtrl
     {
         private SessionCtrl SesCtrl = new SessionCtrl();
-
-        public User CreateUser(DALContext ctx, string Firstname, string Lastname, string Email, string Password)
+        private UserDB uDB = new UserDB();
```
Comment (Arcanedarkness) [inline Backend/Backend/BLL/UserCtrl.cs]: IUserDB should be used here

File: Backend/Backend/DAL/UserDB.cs [hunk 7]
```diff
@@ -45,8 +60,22 @@ public User FindByID(int id)
...
+                    try
+                    {
+                        ctx.Entry(entity).State = System.Data.Entity.EntityState.Modified;
+                        ctx.SaveChanges();
+                        ctxTransaction.Commit();
+                    }
+                    catch (Exception)
+                    {
+
+                        ctxTransaction.Rollback();
```
Comment (FnuGk) [inline Backend/Backend/DAL/UserDB.cs:80]: throw the exception

**Review by Arcanedarkness — CHANGES_REQUESTED** [review]

**Review by FnuGk — CHANGES_REQUESTED** [review]
Only a few changes. Otherwise it looks fine

**Review by Arcanedarkness — APPROVED** [review]

**Review by FnuGk — APPROVED** [review]
!

**Review by Arcanedarkness — APPROVED** [review]

**Pull request merged**